fs.go

```diff
@@ -770,11 +806,22 @@ func (h *fsHandler) handleRequest(ctx *RequestCtx) {
 		return
 	}
 
-	r, err := ff.NewReader()
-	if err != nil {
-		ctx.Logger().Printf("cannot obtain file reader for path=%q: %s", path, err)
-		ctx.Error("Internal Server Error", StatusInternalServerError)
-		return
+	var r io.Reader
+	if cache {
+		r, err = ff.NewBufferReader()
+		if err != nil {
+			ctx.Logger().Printf("error reading buffer from %q: %s", ff.f.Name(), err)
```

Comment:
Having and empty file and caching on will now always print this error. Having an empty file should work normally.